## Upgrade the safety module to v1.5 PART 1

_Updated as of block [17173623](https://etherscan.io/block/17173623) at 5/2/2023, 10:16:11 AM ET_

- ID: 214
- Proposer: [0x329c54289Ff5D6B7b7daE13592C6B1EDA1543eD4](https://etherscan.io/address/0x329c54289Ff5D6B7b7daE13592C6B1EDA1543eD4)
- Start Block: 17176895 (5/2/2023, 10:05:07 PM ET)
- End Block: 17240895 (5/12/2023, 1:11:47 PM ET)
- Targets: [0xe427FCbD54169136391cfEDf68E96abB13dA87A0](https://etherscan.io/address/0xe427FCbD54169136391cfEDf68E96abB13dA87A0#code)
- Executor: [0x79426A1c24B2978D90d7A5070a46C65B07bC4299](https://etherscan.io/address/0x79426A1c24B2978D90d7A5070a46C65B07bC4299) (unknown executor)
- Simulation: [https://dashboard.tenderly.co/me/simulator/3cd69820-3f6f-46a9-8142-927486a49301](https://dashboard.tenderly.co/me/simulator/3cd69820-3f6f-46a9-8142-927486a49301)

<details>
  <summary>Proposal text</summary>

# Summary

This AIP presents the community with the opportunity to upgrade the safety module to v1.5 which introduces:

- a new slashing mechanism
- new cooldown mechanism
- a transfer hook to be used by GHO

The new version also adds some convenience features like permit support or the ability to claim and stake in a single transaction.

# Motivation

On September 2020, the Aave Safety Module was introduced into the ecosystem, to improve the protection of the liquidity protocol, adding an extra utility for the AAVE token: AAVE or AAVE/WETH BPT holders stake their assets to act as a defensive layer in front of any shortfall event.

Since then, apart from really minor upgrades, the contracts have remained the same and in parallel, the system has been adopted by other communities.

As part of our engagement with Aave, we identified that the Safety Module is a clear area of improvement in the ecosystem, from 2 different standpoints:

- Technical. Improving the basic existing mechanisms of the contracts, but without disrupting radically the current design of the SM.
- Conceptual. Making more efficient the SM dynamics (e.g. slashing, rewards distribution rules, etc), by modifying the whole design.

## New slashing mechanism

On the running SM v1, in order to slash, an ad-hoc governance proposal is required, involving important development overhead, which is not ideal.

The new SM v1.5 adds an enhanced mechanism to facilitate slashing of the underlying by tracking an exchange rate between the staked AAVE and the stkAAVE received by stakers. The mechanism is simple: when somebody stakes AAVE, they receive a certain amount of stkAAVE, which no longer is 1:1 equivalent, as it keeps track of slashing (meaning with the same stkAAVE, stakers are able to claim less AAVE).

## New cooldown mechanism

The current cooldown on SM v1 consists of a time delay of 10 days to be respected whenever anybody wants to redeem the staked AAVE. At the moment, this cooldown is affected by in/outflows of stkAAVE, both via transfer() and stake()/redeem(), in order to protect the system from being gamed. Even with this protection mechanism the mechanism is currently gamable, by staking aave within the active cooldown window to [extend the window](https://github.com/bgd-labs/aave-stk-v1-5/issues/6).

This mechanism is not really optimal and adds important complexity, so on v1.5 has been changed to the following: after activation of cooldown, a staker will be able to redeem the minimum balance he will hold between cooldown activation and redeem window. Apart from the user-level cooldown new mechanics, we also propose to increase the cooldown period from 10 days to 20 days.

## GHO Transfer hook

The Aave community has already approved the deployment and activation of the GHO stablecoin and the first facilitator will be the Aave v3 Ethereum pool.

In order to enable the discount mechanism by holding stkAAVE described on the GHO proposal, the design of Aave Companies requires to introduce of a piece of logic on the stkAAVE transfer(), in order to “notify” the GHO facilitator system of stakers’ balances.

We have evaluated this and we think it is acceptable, so it will be included in this upgrade. From a technical perspective, stkAAVE includes logic protections to remain unaffected if anything would go wrong with the GHO facilitator, which at the same time should be considered a trustable entity, as it will be controlled by the Aave governance.

## Misc

The upgrade also allowed us to add some smaller ux improvements:

- a new `stakeWithPermit()` is introduced on `stkAAVE`

- convenience methods to batch claiming and staking/redeeming into a single transaction have been added as `claimRewardsAndRedeem` and `claimRewardsAndStake` have been added to `stkAAVE`

- `preview*()` methods have been added to follow the `4626` standard more closely, even if there's no objective to be compliant

# Specification PART 1

The proposal is split in two parts, as currently the `stkAAVE` is controlled by the `LONG_EXECUTOR` where `stkABPT` is controlled by the `SHORT_EXECUTOR`. As there's a single executor per proposal `Part 1` upgrades the `stkAAVE` implementation and targets the `LONG_EXECUTOR`. `Part 2` upgrades the `stkABPT` implementation and targets the `SHORT_EXECUTOR`.

The proposal will:

- transfer ownership of `stkAAVE` to a `ProxyAdmin` controlled by the `LONG_EXECUTOR`. This is done for consistency reasons with `stkABPT` as the `SHORT_EXECUTOR` cannot be the `SLASHING_ADMIN` and `PROXY_ADMIN` at the same time.

```solidity
IInitializableAdminUpgradeabilityProxy(STK_AAVE).changeAdmin(
  address(AaveMisc.PROXY_ADMIN_ETHEREUM_LONG)
);
```

- deploy the new implementation while maintaining the current values:

```solidity
StakedAaveV3 newImpl = new StakedAaveV3(
  IERC20(AaveV2EthereumAssets.AAVE_UNDERLYING),
  IERC20(AaveV2EthereumAssets.AAVE_UNDERLYING),
  GenericProposal.UNSTAKE_WINDOW,
  GenericProposal.REWARDS_VAULT,
  GenericProposal.EMISSION_MANAGER,
  GenericProposal.DISTRIBUTION_DURATION
);
```

- upgrade `stkAAVE` implementation. The `SLASHING_ADMIN`, `COOLDOWN_ADMIN` and `CLAIM_HELPER` are all initialized as the `SHORT_EXECUTOR`. `MAX_SLASHING` is set to `30%`, `COOLDOWN_SECONDS` is increased to `20 days`:

```solidity
ProxyAdmin(AaveMisc.PROXY_ADMIN_ETHEREUM_LONG).upgradeAndCall(
  TransparentUpgradeableProxy(payable(STK_AAVE)),
  address(newImpl),
  abi.encodeWithSignature(
    'initialize(address,address,address,uint256,uint256)',
    GenericProposal.SLASHING_ADMIN,
    GenericProposal.COOLDOWN_ADMIN,
    GenericProposal.CLAIM_HELPER,
    GenericProposal.MAX_SLASHING,
    GenericProposal.COOLDOWN_SECONDS
  )
);
```

# References

A list of relevant links like for this proposal:

- [ProposalPayloads](https://github.com/bgd-labs/aave-stk-v1-5/blob/main/src/contracts/ProposalPayload.sol)
- [StakedTokenV3](https://github.com/bgd-labs/aave-stk-v1-5/blob/main/src/contracts/StakedTokenV3.sol)
- [StakedAaveV3](https://github.com/bgd-labs/aave-stk-v1-5/blob/main/src/contracts/StakedAaveV3.sol)
- [sigma prime audit](https://github.com/bgd-labs/aave-stk-v1-5/blob/main/audits/Sigma_Prime_Aave_Safety_Module_Security_Assessment_Report_v2.pdf)
- [Certora audit](https://github.com/bgd-labs/aave-stk-v1-5/blob/main/audits/Certora_FV_Report.pdf)
- [Certora specs](https://github.com/bgd-labs/aave-stk-v1-5/tree/main/certora/specs)
- [Test Cases](https://github.com/bgd-labs/aave-stk-v1-5/tree/main/tests)
- [Technical review by aave companies](https://governance.aave.com/t/technical-review-aave-safety-module-v1-5/12436)

# Copyright

Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).

</details>

### Checks

#### Reports all state changes from the proposal ✅ Passed

Info:

- State changes:

```diff
# InitializableAdminUpgradeabilityProxy at `0x4da27a545c0c5B758a6BA100e3a049001de870f5` with implementation unknown contract name at `0xE2E8Badc5d50f8a6188577B89f50701cDE2D4e19`
@@ lastInitializedRevision @@
- 4
+ 5
@@ Slot `0x000000000000000000000000000000000000000000000000000000000000004f` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x00000000000000000000000000000000000000000000000000000000001a5e00"
@@ Slot `0x0000000000000000000000000000000000000000000000000000000000000050` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x0000000000000000000000000000000000000000000000000000000000000bb8"
@@ Slot `0x0000000000000000000000000000000000000000000000000000000000000051` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x0000000100000000000000000000000000000000000000000de0b6b3a7640000"
@@ Slot `0x2d3d1c9680bbdec853995b6859c48c938bd9c76a8e98a938491cbf18b6665bed` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x000000000000000000000000ee56e2b3d491590b5b31738cc34d5232f378a8d5"
@@ Slot `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc` @@
- "0x000000000000000000000000e2e8badc5d50f8a6188577b89f50701cde2d4e19"
+ "0x000000000000000000000000aa9faa887bce5182c39f68ac46c43f36723c395b"
@@ Slot `0xae7f735405eae778c062be95d86a15d8e443f03071caf36c523900c28f6324ad` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x000000000000000000000000000000000000000de0b6b3a76400000001071341"
@@ Slot `0xb476eedba4f8c885c12cb813443bb78e21239c276d1586e24afd457c6ea531b7` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x000000000000000000000000ee56e2b3d491590b5b31738cc34d5232f378a8d5"
@@ Slot `0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103` @@
- "0x00000000000000000000000079426a1c24b2978d90d7a5070a46c65b07bc4299"
+ "0x00000000000000000000000086c3ffee349a7cff7ca88c449717b1b133bfb517"
@@ Slot `0xcf23cc7089fed28926e3cca2bfa1d6811ea4ce7e8c64119a740cf101b353080f` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x000000000000000000000000ee56e2b3d491590b5b31738cc34d5232f378a8d5"
```

```diff
# InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`
@@ `_allowances` key `0x4da27a545c0c5b758a6ba100e3a049001de870f5`.0x4da27a545c0c5b758a6ba100e3a049001de870f5 @@
- 0
+ 115792089237316195423570985008687907853269984665640564039457584007913129639935

```

```diff
# unknown contract name at `0xAa9FAa887bce5182C39F68Ac46C43F36723C395b`
@@ Slot `0x0000000000000000000000000000000000000000000000000000000000000009` @@
- "0x0000000000000000000000000000000000000000000000000000000000000000"
+ "0x0000000000000000000000000000000000000000000000000000000000000005"
```

#### Check stack trace of the proposal ✅ Passed

Info:

- There is no SELFDESTRUCT inside of delegated call

#### Reports all events emitted from the proposal ✅ Passed

Info:

- Events Emitted:
  - InitializableAdminUpgradeabilityProxy at `0x4da27a545c0c5B758a6BA100e3a049001de870f5` with implementation unknown contract name at `0xE2E8Badc5d50f8a6188577B89f50701cDE2D4e19`
    - `AdminChanged(previousAdmin: 0x79426a1c24b2978d90d7a5070a46c65b07bc4299, newAdmin: 0x86c3ffee349a7cff7ca88c449717b1b133bfb517)`
    - `Upgraded(implementation: 0xaa9faa887bce5182c39f68ac46c43f36723c395b)`
    - `RoleClaimed(newAdmin: 0xee56e2b3d491590b5b31738cc34d5232f378a8d5, role: 0)`
    - `RoleClaimed(newAdmin: 0xee56e2b3d491590b5b31738cc34d5232f378a8d5, role: 1)`
    - `RoleClaimed(newAdmin: 0xee56e2b3d491590b5b31738cc34d5232f378a8d5, role: 2)`
    - `MaxSlashablePercentageChanged(newPercentage: 3000)`
    - `CooldownSecondsChanged(cooldownSeconds: 1728000)`
    - `ExchangeRateChanged(exchangeRate: 1000000000000000000)`
  - InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`
    - `Approval(owner: 0x4da27a545c0c5b758a6ba100e3a049001de870f5, spender: 0x4da27a545c0c5b758a6ba100e3a049001de870f5, value: 115792089237316195423570985008687907853269984665640564039457584007913129639935)`

#### Check all targets are verified on Etherscan ✅ Passed

Info:

- Targets:
  - 0xe427FCbD54169136391cfEDf68E96abB13dA87A0: Contract (verified) (ProposalPayloadStkAave)

#### Check all touched contracts are verified on Etherscan ✅ Passed

Info:

- Touched address:
  - 0xD73a92Be73EfbFcF3854433A5FcbAbF9c1316073: EOA (verification not applicable)
  - 0xEC568fffba86c094cf06b22134B23074DFE2252c: Contract (verified) (AaveGovernanceV2)
  - 0x79426A1c24B2978D90d7A5070a46C65B07bC4299: Contract (verified) (Executor)
  - 0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e: Contract (verified) (GovernanceStrategy)
  - 0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9: Contract (verified) (InitializableAdminUpgradeabilityProxy)
  - 0x96F68837877fd0414B55050c9e794AECdBcfCA59: Contract (verified) (AaveTokenV2)
  - 0xe427FCbD54169136391cfEDf68E96abB13dA87A0: Contract (verified) (ProposalPayloadStkAave)
  - 0x4da27a545c0c5B758a6BA100e3a049001de870f5: Contract (verified) (InitializableAdminUpgradeabilityProxy)
  - 0xAa9FAa887bce5182C39F68Ac46C43F36723C395b: Contract (not verified)
  - 0x86C3FfeE349A7cFf7cA88C449717B1b133bfb517: Contract (not verified)

#### Runs solc against the verified contracts ✅ Passed

Info:

-

<details>
<summary>View Details</summary>
<details>
<summary>View warnings for InitializableAdminUpgradeabilityProxy at `0x4da27a545c0c5B758a6BA100e3a049001de870f5` with implementation unknown contract name at `0xE2E8Badc5d50f8a6188577B89f50701cDE2D4e19`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.12' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.12`.

```

</details>

<details>
<summary>View warnings for InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.

```

</details>

<details>
<summary>View warnings for AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

```

</details>

<details>
<summary>View warnings for GovernanceStrategy at `0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

```

</details>

<details>
<summary>View warnings for ProposalPayloadStkAave at `0xe427FCbD54169136391cfEDf68E96abB13dA87A0`</summary>

```
INFO:CryticCompile:solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
ERROR:CryticCompile:Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.8.17' not installed (set by SOLC_VERSION). Run `solc-select install 0.8.17`.

```

</details>

</details>

#### Runs slither against the verified contracts ✅ Passed

Info:

-

<details>
<summary>View Details</summary>

<details>
<summary>Slither report for InitializableAdminUpgradeabilityProxy at `0x4da27a545c0c5B758a6BA100e3a049001de870f5` with implementation unknown contract name at `0xE2E8Badc5d50f8a6188577B89f50701cDE2D4e19`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.12' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.12`.

Error in 0x4da27a545c0c5B758a6BA100e3a049001de870f5
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.12' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.12`.


```

</details>

<details>
<summary>Slither report for InitializableAdminUpgradeabilityProxy at `0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9` with implementation AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.

Error in 0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.6.10' not installed (set by SOLC_VERSION). Run `solc-select install 0.6.10`.


```

</details>

<details>
<summary>Slither report for AaveTokenV2 at `0x96F68837877fd0414B55050c9e794AECdBcfCA59`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

Error in 0x96F68837877fd0414B55050c9e794AECdBcfCA59
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.


```

</details>

<details>
<summary>Slither report for GovernanceStrategy at `0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.

Error in 0xb7e383ef9B1E9189Fc0F71fb30af8aa14377429e
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.7.5' not installed (set by SOLC_VERSION). Run `solc-select install 0.7.5`.


```

</details>

<details>
<summary>Slither report for ProposalPayloadStkAave at `0xe427FCbD54169136391cfEDf68E96abB13dA87A0`</summary>

```
solc-select is not installed.
Run "pip install solc-select" to enable automatic switch of solc versions
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.8.17' not installed (set by SOLC_VERSION). Run `solc-select install 0.8.17`.

Error in 0xe427FCbD54169136391cfEDf68E96abB13dA87A0
Traceback (most recent call last):
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 163, in run_solc_standard_json
    solc_json_output = json.loads(stdout)
                       ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.1/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 834, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/slither-analyzer/0.9.2/libexec/lib/python3.11/site-packages/slither/__main__.py", line 87, in process_all
    compilations = compile_all(target, **vars(args))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 620, in compile_all
    compilations.append(CryticCompile(target, **kwargs))
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 110, in __init__
    self._compile(**kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/crytic_compile.py", line 530, in _compile
    self._platform.compile(self, **kwargs)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/etherscan.py", line 362, in compile
    solc_standard_json.standalone_compile(filenames, compilation_unit, working_dir=working_dir)
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 66, in standalone_compile
    targets_json = run_solc_standard_json(
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/opt/crytic-compile/libexec/lib/python3.11/site-packages/crytic_compile/platform/solc_standard_json.py", line 193, in run_solc_standard_json
    raise InvalidCompilation(f"Invalid solc compilation {stderr}")
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/solc-select/1.0.2/libexec/lib/python3.11/site-packages/solc_select/solc_select.py", line 51, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: Version '0.8.17' not installed (set by SOLC_VERSION). Run `solc-select install 0.8.17`.


```

</details>

</details>
